src/adapter/auth-oso-rebac/src/oso_resource_service_impl.rs

```diff
@@ -84,48 +42,18 @@ impl OsoResourceServiceImpl {
             return Ok(UserActor::anonymous());
         };
 
-        // First, an attempt to get from the cache
-        {
-            let readable_state = self.state_holder.state.read().await;
-
-            let account_id_stack = account_id.as_did_str().to_stack_string();
-            let maybe_cached_user_actor = readable_state
-                .user_actor_cache_map
-                .get(account_id_stack.as_str())
-                .cloned();
-
-            if let Some(cached_user_actor) = maybe_cached_user_actor {
-                return Ok(cached_user_actor);
-            }
-        }
-
-        // The second attempt is from the database
-        let user_actor = {
-            let account = match self.account_service.get_account_by_id(account_id).await {
-                Ok(found_account) => found_account,
-                Err(e) => return Err(e.into()),
-            };
-
-            // TODO: Private Datasets: absorb the `is_admin` attribute
-            //       from the Accounts domain
-            //       https://github.com/kamu-data/kamu-cli/issues/766
-            // let account_properties = self
-            //     .rebac_service
-            //     .get_account_properties(&account.id)
-            //     .await
-            //     .int_err()?;
-
-            UserActor::logged(&account.id, account.is_admin)
+        let account = match self.account_service.get_account_by_id(account_id).await {
+            Ok(found_account) => found_account,
+            Err(e) => return Err(e.into()),
```

Comment:
Previously, we requested `account` to read `account.is_admin` from it. 

At this point, it looks like we can remove the `account_service.get_account_by_id()` call at all.

In that case, `GetUserActorError::NotFound` will no longer be needed either.